Sources/StructuralCore/Structural.swift

```diff
@@ -21,103 +74,187 @@ public protocol Structural {
     init(structuralRepresentation: StructuralRepresentation)
 
     /// A structural representation of `self`.
-    var structuralRepresentation: StructuralRepresentation { get set }
+    var structuralRepresentation: StructuralRepresentation {
+        get set
+    }
 }
 
-/// Structural representation of a Swift struct.
+/// The structural representation of a Swift struct type.
+///
+/// `StructuralStruct<...>` is the compiler-synthesized
+/// `StructuralRepresentation` for structs conforming to `Structural`.
+///
+/// - Parameter Properties: a type representing the names and values of stored
+///   properties.
 public struct StructuralStruct<Properties> {
-    public var type: Any.Type?
-    public var properties: Properties
+    /// The type of the represented struct type.
+    public var representedType: Any.Type?
+    /// The names and values of stored properties.
+    public var body: Properties
```

Comment:
I wonder why all properties in this file are `var` instead of `let`? Are there use cases for `var`?

Reply:
The use case is to allow easy in-place mutation of the `StructuralRepresentation` (see `MyInplaceAdd` protocol for an example). Otherwise if make everything immutable, users will have to do a deep copy of the whole representation every time you change a single part of it.

Reply:
I see, thanks!

I think in-place mutation is a good reason for making `StructuralStruct.body` and `StructuralEnum.body` be mutable.

How about `var StructuralProperty.name` (mutable property name) and `var StructuralProperty.isMutable` (mutable property mutability) - are there use cases for those? Even if not, keeping those as `var` sounds good to me for now, in case there are future use cases.

Reply:
Changes to `name` and `isMutable` would not have any effect, so it makes sense to make them `let` instead.